lib/bin.js

```diff
@@ -11,12 +11,21 @@ function bin (args, silent, cb) {
     silent = false
   }
   var b = npm.bin
-  var PATH = osenv.path()
 
   if (!silent) console.log(b)
   process.nextTick(cb.bind(this, null, b))
+}
 
-  if (npm.config.get('global') && PATH.indexOf(b) === -1) {
-    npm.config.get('logstream').write('(not in PATH env variable)\n')
-  }
+bin.ensureInPath = function ensureInPath (cb) {
```

Comment:
So the first thing I'd do is write a functional test for this part. To do that you'll want to be able to control all of the variables that go into this:
```
var requireInject = require('require-inject')
var binPath = '/usr/local/bin/npm' // or whatever
var envPATH = '/usr/bin:/bin:/usr/local/bin'
var isGlobal = true
var bin = requireInject('../../lib/bin.js', {
  npm: {
    bin: binPath,
    config: {
      get: function (key) {
          if (key !== 'global') throw new Error('unexpected config key: ' + key)
          return isGlobal
      }
    }
  },
  osenv: {
    path: function () {
      return envPATH
    }
  }
})